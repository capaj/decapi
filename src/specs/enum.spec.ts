import { graphql } from 'graphql'
import {
  registerEnum,
  ObjectType,
  Field,
  Arg,
  SchemaRoot,
  Query,
  compileSchema
} from '../index'

export enum TestEnum {
  Foo = 'Foo',
  Bar = 'Bar',
  Baz = 'Baz'
}

registerEnum(TestEnum, { name: 'TestEnum' })

@ObjectType()
class Hello {
  @Field()
  world(@Arg({ type: TestEnum }) name: TestEnum): string {
    return `Hello, ${name}`
  }
}

@SchemaRoot()
class FooSchema {
  @Query()
  hello(): Hello {
    return new Hello()
  }
}

const schema = compileSchema(FooSchema)

describe('Query with enums', () => {
  it('Will guard proper enum values', async () => {
    const result = await graphql(
      schema,
      `
        {
          hello {
            world(name: Foob)
          }
        }
      `
    )

    expect(result.errors).toBeDefined()
    expect(result.errors).toMatchInlineSnapshot(`
      Array [
        [GraphQLError: Expected type TestEnum, found Foob. Did you mean the enum value Foo?],
      ]
    `)
  })
})
