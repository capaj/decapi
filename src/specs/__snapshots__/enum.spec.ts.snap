// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query with enums Will guard proper enum values 1`] = `
Array [
  [GraphQLError: Expected type TestEnum, found Foob; Did you mean the enum value Foo?],
]
`;
